pulsar-broker/src/main/java/org/apache/pulsar/broker/service/persistent/PersistentTopic.java

```diff
@@ -429,6 +431,25 @@ public AtomicLong getPendingWriteOps() {
         return pendingWriteOps;
     }
 
+    public CompletableFuture<Void> unloadSubscription(String subName) {
+        final PersistentSubscription sub = subscriptions.get(subName);
+        if (sub == null) {
+            return CompletableFuture.failedFuture(new RestException(Response.Status.NOT_FOUND,
+                    String.format("Subscription %s not found", subName)));
+        }
+        if (Compactor.COMPACTION_SUBSCRIPTION.equals(sub.getName())){
+            return CompletableFuture.failedFuture(new RestException(Response.Status.BAD_REQUEST,
+                    "Could not reload the compaction subscription"));
+        }
+        // Fence old subscription -> Rewind cursor -> Replace with a new subscription.
+        return sub.disconnect().thenAccept(ignore -> {
+            sub.getCursor().rewind();
```

Comment:
It looks like `unloadSubscription` just rewinds the cursor. Why we can't close it directly like `unloadTopic`?

Reply:
`unloadTopic` will close the `ManagedLedger` and close all cursors, which guarantees that the states of ML and cursors are consistent. Only closing one cursor under the ML will affect the work of ML, which is a dangerous behavior.

Reply:
Therefore, is calling this behaviour "rewind cursor" more suitable?

Reply:
No, this operation did this two things:
- recreate the subscription: This will clear all attributes of the Subscription, which will help resolve the problem of Subscription due to incorrect attributes quickly.
- rewind cursor
We can't just call it `rewindCursor`